lua/harpoon/mark.lua

```diff
@@ -31,6 +31,25 @@ local function get_buf_name(id)
     return ""
 end
 
+local function mark_exists(buf_name)
+    local marks = harpoon.get_mark_config().marks
+
+    for idx = 1, M.get_length() do
+        if marks[idx] == buf_name then
+            return true
+        end
+    end
+
+    return false
+end
+
+local function validate_buf_name(buf_name)
+    if buf_name == "" or buf_name == nil then
+        error("Couldn't find a valid file name to mark, sorry.")
```

Comment:
We do need to come up with a logging strategy.  We don't want to over due it to the end user, but I do like it during alpha!  

We can revisit this later.